### DISCUSSION TITLE
Recommeneded basic smart switch to run Tasmota

### DISCUSSION THREAD

**Discussion by apastuszak** [arendst/Tasmota#19480] [General]
Reading through the open threads here, I see a disdain for the venerable Sonoff Basic. So, I am curious. If you want a simple on/off smart switch you can wire into an extension cord and flash Tasmota onto, what is the current recommendation?

I ordered some Shelly Plus 1 ULs today, but they seem to have overheating issues for some people.

Sonoff Basic's a nice and cheap. But I like the fact that the Shelly is UL listed. I will entertain other options.

**Comment by lalo-uy** [comment]
Do you really need the full featured Tasmota on these devices?
Why not compile a trimmed and less power hungry version?

**Comment by Wintespe** [comment]
@apastuszak Yes, that is my impression too. The current Tamota versions have a lot of features that are not needed for such a simple application.
However, sometimes a single feature is needed for certain applications. In this case a minimal configuration is not possible.
I simply use an older version of Tasmota. I can modify it to my needs and then compile it.

**Comment by apastuszak** [comment]
Really I just need it to work with Home Assistant, so an older firmware is fine. I'm just looking for reliability. In another thread I started, some people claimed than any smart switch is better that a Sonoff Basic R2. So, I'm curious what other are using instead of a Sonoff Basic. I like the Basic R3, because I can get Tasmota on it without needing to use a serial adapter.

**Reply by barbudor** [reply]
Why not using tasmota-lite ?
That's what I'm running on my Sonoff basic R1

**Reply by apastuszak** [reply]
That's fine. But it still doesn't answer the question of what smart switch is more reliable than a Sonoff Basic.

**Reply by barbudor** [reply]
In different form factor Shellies are great

I only have personnal experience with Sonoff Basic R1, Shellies and DIN rail mountable RocketController

They are a lot listed on the template page: https://templates.blakadder.com/relay.html#switch%20module
You may want to try the Athom equivalent of SOnoff Basic. I have no personnal experience but they generally source correct products: https://templates.blakadder.com/athom_CB01-TAS.html

**Reply by sfromis** [reply]
It is not about what specific one is better than Sonoff Basic, but that device being an "unusually bad" outlier, as evidenced by lots of users having trouble until replacing by "something else".

**Comment by Wintespe** [comment]
I know this kind of thread too. The Sonoff Basic is a stable piece of hardware when used for what it was designed for by ITEAD. Installing more and more complex firmware on it leads to instability. And it is not easy to modify Tasmota-lite. 

I use Sonoff Basic in all versions for several years without any problems.

**Reply by barbudor** [reply]
Why would you modify Tasmota-lite ?
It just provides what the Sonoff need

**Reply by sfromis** [reply]
No, Sonoff Basic is **not** a stable piece of hardware, they tend to break with age, instead of this issue of not being able to cope with Tasmota. Personally, I've had two break suddenly, not even flashable. And I'm not the only one.

**Comment by Wintespe** [comment]
This is getting a little acadmic. I once intended to use the PID controller in a Sonoff Basic. It didn't work, and I had to use more sophisticated hardware.

**Reply by barbudor** [reply]
PID may be
I'm running a room heater with basic rules, receiving the room temp from a Mija BLE sensor through my BLE gateway

**Comment by apastuszak** [comment]
So what are you using instead of Sonoff Basic?Sent from my iPhoneOn Sep 8, 2023, at 6:53 PM, sfromis ***@***.***> wrote:﻿
No, Sonoff Basic is not a stable piece of hardware, they tend to break with age, instead of this issue of not being able to cope with Tasmota. Personally, I've had two break suddenly, not even flashable. And I'm not the only one.
